File: smart_farm/src/api/aiRecognitionRecord.ts
import { get, http } from '@/utils/request'
import type { PageInfo } from '@/api/user'

export interface AiRecognitionRecord {
  id: number
  userId: number
  username?: string
  nickname?: string
  sourceType: number
  cropImageId?: number
  cameraId?: number
  cameraName?: string
  plotId?: number
  plotName?: string
  plotCode?: string
  captureId?: number
  imageUrl: string
  thumbnailUrl?: string
  imageSize?: number
  status: number
  recognitionStartTime?: string
  recognitionEndTime?: string
  remark?: string
  createTime?: string
  updateTime?: string
  resultId?: number
  recognitionType?: number
  recognitionTypeName?: string
  cropId?: number
  cropName?: string
  resultName?: string
  resultSummary?: string
  resultDetail?: string
  diseasePestId?: number
  diseasePestName?: string
  confidence?: number
  severityLevel?: number
  suggestion?: string
  modelName?: string
  modelVersion?: string
  resultStatus?: number
  failReason?: string
  recognizeTime?: string
  resultRemark?: string
  resultCreateTime?: string
  resultCount?: number
}

export interface AiRecognitionRecordListParams {
  username?: string
  sourceType?: number
  status?: number
  resultStatus?: number
  cameraName?: string
  plotName?: string
  recognitionType?: number
  resultName?: string
  startDate?: string
  endDate?: string
  pageNum: number
  pageSize: number
}

export type AiRecognitionRecordStatisticsParams = Omit<
  AiRecognitionRecordListParams,
  'pageNum' | 'pageSize'
>

export interface AiRecognitionRecordStatistics {
  totalCount?: number
  manualCount?: number
  autoCount?: number
  pendingCount?: number
  runningCount?: number
  completedCount?: number
  failedRecordCount?: number
  successResultCount?: number
  failedResultCount?: number
  firstCreateTime?: string
  latestCreateTime?: string
}

interface ApiResult<T> {
  code: number
  message: string
  data: T
}

export const listAiRecognitionRecords = (params: AiRecognitionRecordListParams) => {
  return get<ApiResult<PageInfo<AiRecognitionRecord>>>(
    '/ai-recognition-record/list',
    params as unknown as Record<string, unknown>,
  )
}

export const getAiRecognitionRecordStatistics = (params: AiRecognitionRecordStatisticsParams) => {
  return get<ApiResult<AiRecognitionRecordStatistics>>(
    '/ai-recognition-record/statistics',
    params as unknown as Record<string, unknown>,
  )
}

export const getAiRecognitionRecordById = (id: number) => {
  return get<ApiResult<AiRecognitionRecord>>(`/ai-recognition-record/${id}`)
}

export const deleteAiRecognitionRecord = (id: number) => {
  return http<ApiResult<void>>({ url: `/ai-recognition-record/${id}`, method: 'delete' })
}

export const batchDeleteAiRecognitionRecords = (ids: number[]) => {
  return http<ApiResult<number>>({ url: '/ai-recognition-record/batch', method: 'delete', data: ids })
}

File: smart_plant/src/main/java/com/smart_plant/smart_plant/entity/AiRecognitionRecord.java
package com.smart_plant.smart_plant.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AiRecognitionRecord {

    private Long id;

    private Long userId;

    private String username;

    private String nickname;

    private Integer sourceType;

    private Long cropImageId;

    private Long cameraId;

    private String cameraName;

    private Long plotId;

    private String plotName;

    private String plotCode;

    private Long captureId;

    private String imageUrl;

    private String thumbnailUrl;

    private Long imageSize;

    private Integer status;

    private LocalDateTime recognitionStartTime;

    private LocalDateTime recognitionEndTime;

    private String remark;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;

    private Long resultId;

    private Long recognitionType;

    private String recognitionTypeName;

    private Long cropId;

    private String cropName;

    private String resultName;

    private String resultSummary;

    private String resultDetail;

    private Long diseasePestId;

    private String diseasePestName;

    private BigDecimal confidence;

    private Integer severityLevel;

    private String suggestion;

    private String modelName;

    private String modelVersion;

    private Integer resultStatus;

    private String failReason;

    private LocalDateTime recognizeTime;

    private String resultRemark;

    private LocalDateTime resultCreateTime;

    private Integer resultCount;
}
